ts41/query-result.d.ts

```diff
@@ -0,0 +1,66 @@
+import {TakeLast} from '..';
+import {Split} from './split';
+import {Trim} from './trim';
+
+/**
+Matches element types returned from a query.
+
+Use cases:
+- anywhere you'd use `querySelector` or `querySelectorAll`.
+- used on other element-selecting functions that are working based on CSS selectors.
+
+If you'd like to see this integrated into TypeScript, please upvote [this issue](https://github.com/microsoft/TypeScript/issues/29037).
+
+@example
+declare function querySelector<T extends string>(query: T): QueryResult<T>;
+
+const anchor = querySelector('div.banner > a.call-to-action') //-> HTMLAnchorElement
+const div = querySelector('input, div') //-> HTMLInputElement | HTMLDivElement
+const svg = querySelector('circle[cx="150"]') //-> SVGCircleElement
+const button = querySelector('button#buy-now') //-> HTMLButtonElement
+const paragraph = querySelector('section p:first-of-type'); //-> HTMLParagraphElement
+*/
+export type QueryResult<T extends string> = MatchEachElement<GetElementNames<T>>;
```

Comment:
Should be named `QueryResult`.

And the export should be at the bottom.

Reply:
It is named that though? Moved to the end of the file.